Easydict/Swift/Feature/Localization/LanguageState.swift

```diff
@@ -45,3 +45,28 @@ class LanguageState: ObservableObject {
         }
     }
 }
+
+// https://developer.apple.com/library/archive/documentation/MacOSX/Conceptual/BPInternational/LanguageandLocaleIDs/LanguageandLocaleIDs.html
+extension Locale {
+    var languageType: LanguageState.LanguageType? {
+        let languageCode = language.languageCode?.identifier
+        let script = language.script?.identifier
+        let region = language.region?.identifier
+
+        let languageCodeScriptRegion = [languageCode, script, region].compactMap { $0 }.joined(separator: "-")
+        let languageCodeScript = [languageCode, script].compactMap { $0 }.joined(separator: "-")
+        let languageCodeRegion = [languageCode, region].compactMap { $0 }.joined(separator: "-")
+
+        if let languageCode, let type = LanguageState.LanguageType(rawValue: languageCode) {
+            return type
+        } else if let type = LanguageState.LanguageType(rawValue: languageCodeScriptRegion) {
```

Comment:
I think it will look clear to omit the `else`, like

```
        if let languageCode, let type = LanguageState.LanguageType(rawValue: languageCode) {
            return type
        }
        if let type = LanguageState.LanguageType(rawValue: languageCodeScriptRegion) {
            return type
        }
        if let type = LanguageState.LanguageType(rawValue: languageCodeScript) {
            return type
        }
        if let type = LanguageState.LanguageType(rawValue: languageCodeRegion) {
            return type
        }
```